### PR TITLE
Updated tests, modified way user input is handled

### PR DESCRIPTION
**Pull request by pault2k14**

### CODE REVIEW

File: EIAClient.java [hunk 1]
```diff
@@ -21,7 +21,13 @@
     // then runs the lambda function.
     public boolean execute(String []userInput) {
         input = userInput;
-        commands.get(input[0].toLowerCase()).run();
+
+        try {
```
Comment (galxy25) [inline EIAClient.java]: good catch ;-)

File: EIAClient.java [hunk 2]
```diff
@@ -32,76 +38,55 @@ public String getHost() {
...
     }
 
 
     // Connect wrapper to determine if the user entered a password or not.
-    public void connect(String []input) {
+    protected boolean connect(String []input) {
 
         if(input.length == 4) {
-            connectNoPassword(input[1], input[2], input[3]);
+            connectToHost(input[1], input[2], input[3], "");
```
Comment (galxy25) [inline EIAClient.java]: This is a more generic version of the nopassword/withpassword connects?

File: FTPApp.java [hunk 3]
```diff
@@ -72,47 +95,53 @@ public boolean help() {
...
+        try {
+            if(inUseMode.execute(input)) {
+                return true;
             }
 
+            else {
+                return false;
+            }
 
+        } catch (NullPointerException | ArrayIndexOutOfBoundsException e) {
```
Comment (galxy25) [inline FTPApp.java]: nice that we don't have to worry about any more than two possible exceptions for one method.

File: apache-commons-io/docs/org/apache/commons/io/class-use/ByteOrderMark.html [hunk 4]
```diff
@@ -1,285 +0,0 @@
...
-<TR>
-<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
-&nbsp;PREV&nbsp;
-&nbsp;NEXT</FONT></TD>
-<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
-  <A HREF="../../../../../index.html?org/apache/commons/io/\class-useByteOrderMark.html" target="_top"><B>FRAMES</B></A>  &nbsp;
-&nbsp;<A HREF="ByteOrderMark.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
-&nbsp;<SCRIPT type="text/javascript">
-  <!--
-  if(window==top) {
```
Comment (galxy25) [inline apache-commons-io/docs/org/apache/commons/io/class-use/ByteOrderMark.html]: +50 for the awesome gitignore to pare down the massive number of files.

**Comment by galxy25** [comment]
@pault2k14 LGTM.

File: EIAClient.java (hunk 2 above)
Comment (pault2k14) [inline EIAClient.java]: Yeah, I figured it was better to use a wrapper and just one connection function.

**Pull request merged**